x-pack/plugins/ml/public/data_frame/common/request.test.ts

```diff
@@ -36,12 +36,34 @@ describe('Data Frame: Common', () => {
     ];
     const aggs: PivotAggsConfig[] = [
       { agg: PIVOT_SUPPORTED_AGGS.AVG, field: 'the-agg-field', aggName: 'the-agg-label' },
+      {
+        agg: PIVOT_SUPPORTED_AGGS.TRANSACTION_DURATION,
+        field: 'the-transaction-field',
+        aggName: 'the-transaction-label',
+      },
     ];
     const request = getDataFramePreviewRequest('the-index-pattern-title', query, groupBy, aggs);
 
     expect(request).toEqual({
       pivot: {
-        aggregations: { 'the-agg-label': { avg: { field: 'the-agg-field' } } },
+        aggregations: {
+          'the-agg-label': { avg: { field: 'the-agg-field' } },
+          'the-transaction-label': {
+            scripted_metric: {
+              combine_script: 'return state',
+              map_script: "state.ts = doc['the-transaction-field'].value.getMillis();",
```

Comment:
This script does not actually work unless `ts` is a magic key that is unique for every `doc` in every `shard`?

For each `shard` you are setting the the `state.ts` value to the last value seen in that `shard`. This means that you are not actually getting the `min` or the `max` for each value in each shard.

Reply:
Thanks for the feedback! I tried to revamp the script to be more in line with what the [this example](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-metrics-scripted-metric-aggregation.html#search-aggregations-metrics-scripted-metric-aggregation) does to correctly handle the data in the init/map/combine scripts. 

The updated version is in 7e52e97.

```
{
  scripted_metric: {
    init_script: 'state.timestamps = []',
    map_script: `state.timestamps.add(doc['the-transaction-field'].value.getMillis())`,
    combine_script: `
      state.min = Double.POSITIVE_INFINITY;
      state.max = Double.NEGATIVE_INFINITY;
      for (timestamp in state.timestamps) {
      state.min = Math.min(state.min, timestamp);
      state.max = Math.max(state.max, timestamp);
      }
      return state;
      `,
    reduce_script: `
      double min = Double.POSITIVE_INFINITY;
      double max = Double.NEGATIVE_INFINITY;
      for (s in states) {
      min = Math.min(min, s.min);
      max = Math.max(max, s.max);
      }
      return max - min;
      `,
    },
  },
}
```

Reply:
Looks better to me :), one optimization is to simply have `state = [min: Double.POSITIVE_INFINITY, max: Double.NEGATIVE_INFINITY` and have map be
```
if (doc['the-transaction-field'].value.getMillis() > state.max) {
}
state.timestamps.add(doc['the-transaction-field'].value.getMillis())
```

Reply:
Nice, much better without the array/for-loop! I pushed another version. FYI I couldn't use `state = [min: Double.POSITIVE_INFINITY, max: Double.NEGATIVE_INFINITY]` because ES would complain that state cannot be reassigned. Here's the update:

```
{
  scripted_metric: {
    init_script: `
      state.min = Double.POSITIVE_INFINITY;
      state.max = Double.NEGATIVE_INFINITY;
    `,
    map_script: `
      double timestamp = doc['the-transaction-field'].value.getMillis();
      state.min = Math.min(state.min, timestamp);
      state.max = Math.max(state.max, timestamp);
    `,
    combine_script: `return state;`,
    reduce_script: `
      double min = Double.POSITIVE_INFINITY;
      double max = Double.NEGATIVE_INFINITY;
      for (s in states) {
        min = Math.min(min, s.min);
        max = Math.max(max, s.max);
      }
      return max - min;
    `,
  },
}
```